libs/designsystem/modal/src/modal/services/modal.helper.ts

```diff
@@ -175,4 +175,12 @@ export class ModalHelper {
       modal.dismiss();
     });
   }
+
+  private handleBackdropDismiss(config: ModalConfig) {
+    if (config.backdropDismiss !== undefined) {
+      return config.backdropDismiss;
+    }
+
+    return config.flavor === 'compact' || config.interactWithBackground ? false : true;
```

Comment:
I think this should be refactored so these are the first checks. We do not want people to be able to overwrite `backdropDismiss` for compact or `interactWithBackground` scenarios :)